compiler/rustc_data_structures/src/binary_search_util/mod.rs

```diff
@@ -10,41 +10,17 @@ pub fn binary_search_slice<'d, E, K>(data: &'d [E], key_fn: impl Fn(&E) -> K, ke
 where
     K: Ord,
 {
-    let Ok(mid) = data.binary_search_by_key(key, &key_fn) else {
+    let size = data.len();
+    let start = data.partition_point(|x| key_fn(x) < *key);
+    // at this point start either points at the first entry with equal key
+    // or is equal to size in case all elements have smaller keys
```

Comment:
```suggestion
    // At this point `start` either points at the first entry with equal or greater key
    // or is equal to `size` in case all elements have smaller keys
```

Reply:
thank you for the suggestions!